src/ifp/src/InitFloorplan.cc

```diff
@@ -160,32 +160,22 @@ static int divCeil(int dividend, int divisor)
 
 void InitFloorplan::initFloorplan(const odb::Rect& die,
                                   const odb::Rect& core,
-                                  const std::string& site_name)
+                                  const std::vector<odb::dbSite*>& user_sites)
 {
   Rect die_area(snapToMfgGrid(die.xMin()),
                 snapToMfgGrid(die.yMin()),
                 snapToMfgGrid(die.xMax()),
                 snapToMfgGrid(die.yMax()));
   block_->setDieArea(die_area);
-  auto sites = getSites();
```

Comment:
Why remove the auto-detection?  It forces the python API to require the site be named when there is only one.  I think -sites should be for additional sites over what the netlist requires rather than mandatory.

Reply:
The call is there, just changed it to `std::set<odb::dbSite*> sites = ...` (it's still on line 170). I don't think it changes the Python API, it can just pass in an empty list.